urifind.go

```diff
@@ -305,145 +305,26 @@ func TitleGet(url string) (string, string, error) {
 	return title, lastUrl, nil
 }
 
-// parses the incoming HTML fragment and tries to extract text from
-// suitable tags. Currently this is the page’s title tag and tweets
-// when the HTML-code is similar enough to twitter.com. Returns
-// title and tweet.
-func titleParseHtml(body io.Reader, detailedSearch bool) string {
-	z := html.NewTokenizer(body)
-
-	title := ""
-
-	githubDesc := ""
-
-	tweetText := ""
-	tweetUserName := ""
-	tweetUserScreenName := ""
-	tweetPicUrl := ""
-
-	titleDepth := -1
-	tweetPermalinkDepth := -1
-	tweetTextDepth := -1
-	githubDescDepth := -1
-
-	depth := 0
-TokenizerLoop:
-	for {
-		tt := z.Next()
-		tn, hasAttr := z.TagName()
-
-		if bytes.Equal(tn, []byte("img")) {
-			// skip tags that are likely not to be closed. E.g. <img src="asd"> would
-			// permanently increase the depth by one and thus break the simple logic
-			// below.
-			continue
+func extractTitleFromHTML(body io.Reader) (string, error) {
+	node, err := html.Parse(body)
+	if err != nil {
+		return "", err
+	}
+	var title string
+	var f func(*html.Node)
+	f = func(n *html.Node) {
+		if n.Type == html.ElementNode && n.Data == "title" {
+			title = n.FirstChild.Data
```

Comment:
Handlet das quotes sinnvoll? i.e. der title-tag von twitter sieht bei mir so aus:
Justin Roiland on Twitter: &quot;My @vrlosangeles #VRLA keynote is this Saturday! Get ready for THE STATE OF ADDRESS OF VR! https://t.co/jbJuJfhaky&quot